fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogBinding.kt

```diff
@@ -24,17 +27,60 @@ class TranslationsDialogBinding(
     private val getTranslatedPageTitle: (localizedFrom: String?, localizedTo: String?) -> String,
 ) : AbstractBinding<BrowserState>(browserStore) {
 
+    @Suppress("LongMethod")
     override suspend fun onState(flow: Flow<BrowserState>) {
-        flow.mapNotNull { state -> state.findTab(sessionId) }
+        // Browser level flows
+        val browserFlow = flow.mapNotNull { state -> state }
+            .distinctUntilChangedBy {
+                it.translationEngine
+            }
+
+        // Session level flows
+        val sessionFlow = flow.mapNotNull { state -> state.findTab(sessionId) }
             .distinctUntilChangedBy {
                 it.translationsState
             }
-            .collect { sessionState ->
-                val translationsState = sessionState.translationsState
 
+        // Applying the flows together
+        sessionFlow
+            .combine(browserFlow) { sessionState, browserState -> TranslationsFlowState(sessionState, browserState) }
+            .collect {
+                    state ->
+
+                // Browser Translations State Behavior (Global)
+                val browserTranslationsState = state.browserState.translationEngine
+                val translateFromLanguages =
+                    browserTranslationsState.supportedLanguages?.fromLanguages
+                translateFromLanguages?.let {
+                    translationsDialogStore.dispatch(
+                        TranslationsDialogAction.UpdateTranslateFromLanguages(
+                            translateFromLanguages,
+                        ),
+                    )
+                }
```

Comment:
```suggestion
                val translateFromLanguages =
                    browserTranslationsState.supportedLanguages?.fromLanguages?.let {
                    translationsDialogStore.dispatch(
                        TranslationsDialogAction.UpdateTranslateFromLanguages(
                            it,
                        ),
                    )
                }
```
You can also do the suggested you wanted to.

Reply:
Thanks! The value itself was also needed much further below, so left it the same